modules/core/src/main/scala/ch/epfl/bluebrain/nexus/iam/core/identity/Identity.scala

```diff
@@ -52,4 +53,21 @@ object Identity {
 
   implicit val identityShow: Show[Identity] = Show.fromToString[Identity]
 
+  implicit val identityDecoder: Decoder[Identity] =
+    Decoder.forProduct3[Option[String], Option[String], Option[String], Identity]("origin", "subject", "group") {
+      case (None, _, _)                      => Anonymous
+      case (Some(origin), None, None)        => AuthenticatedRef(Uri(origin))
+      case (Some(origin), None, Some(group)) => GroupRef(Uri(origin), group)
```

Comment:
`Uri(origin)` can throw, the decoder instance should return a `Left[_]` rather than propagating the exception.